src/secret_sharing/malicious_replicated.rs

```diff
@@ -49,26 +46,30 @@ impl<F: Field> MaliciousReplicated<F> {
     }
 }
 
-impl<F: Field> Add for MaliciousReplicated<F> {
-    type Output = Self;
+impl<F: Field> Add<Self> for &MaliciousReplicated<F> {
+    type Output = MaliciousReplicated<F>;
 
-    fn add(self, rhs: Self) -> Self {
-        Self {
-            x: self.x + rhs.x,
-            rx: self.rx + rhs.rx,
+    fn add(self, rhs: Self) -> Self::Output {
+        MaliciousReplicated {
+            x: &self.x + &rhs.x,
+            rx: &self.rx + &rhs.rx,
         }
     }
 }
 
-impl<F: Field> AddAssign for MaliciousReplicated<F> {
-    fn add_assign(&mut self, rhs: Self) {
-        *self = self.add(rhs);
+impl<F: Field> Add<&Self> for MaliciousReplicated<F> {
```

Comment:
This one lets you add a reference to a value.  Note that it modifies the value in place to avoid an allocation.

Note that there is no way to add a value to a value, though there is no real reason not to have it, it creates a good incentive to take values by reference everywhere, because passing a value doesn't work.